src/android/device.rs

```diff
@@ -23,6 +23,102 @@ fn gradlew(config: &Config, env: &Env) -> bossy::Command {
         .with_arg(config.project_dir())
 }
 
+#[cfg(not(target_os = "macos"))]
+mod BundleToolJar {
+    use std::path::PathBuf;
+
+    use crate::android::device::{Config, Invalidutf8Error};
+
+    pub struct BundletoolJarInfo {
+        version: &'static str,
+    }
+
+    impl BundletoolJarInfo {
+        const fn new(version: &'static str) -> Self {
+            Self { version }
+        }
+
+        fn jar(&self) -> String {
+            format!("bundletool-all-{}.jar", self.version)
+        }
+
+        pub fn jar_path(&self, config: &Config) -> PathBuf {
+            config.project_dir().join(self.jar())
+        }
+
+        pub fn download_url(&self) -> String {
+            format!(
+                "https://github.com/google/bundletool/releases/download/{}/{}",
+                self.version,
+                self.jar()
+            )
+        }
+
+        pub fn run_command(&self, config: &Config) -> Result<String, Invalidutf8Error> {
```

Comment:
if you return `bossy::Command` instead, you can save yourself both the string conversion and the need for `Result`